### PR TITLE
Use a component to render NULL values

### PR DESCRIPTION
**Pull request by seancolsen**
In our [discussion about how to display `NULL` to the user](https://github.com/centerofci/mathesar/discussions/778), we had support for using a component to keep the styling consistent. This PR replaces all the instances of "null" I could find in the UI with a new component.

## Notes

- ~Without #766, I didn't have an appropriate directory in which to put this component. So I stuck it in `mathesar-components` for now but we'll need to change that when merging this work with the work in 766.~ **Resolved**

- I recognize that our terminology is still up for debate. This PR sticks with the precedent we have already of using `null` in the UI (though it does change it to be in uppercase.)

- I also made some stylistic choices on my own here. Happy to adjust them within this PR or later.

    | Before | After |
    | --- | --- |
    | image | image |

- It's worth noting that before this PR the import preview did not stylize null values at all, giving the user no indication that a string containing "null" would be any different than an actual `NULL`.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by kgodey** [comment]
I like the styling :)

> Without Rename components dir to component-library #766, I didn't have an appropriate directory in which to put this component.

Why not branch this PR off of the branch in #766? Once #766 is merged, then this PR will auto update to be off of `master`. You can mark it as blocked until then.

**Comment by seancolsen** [comment]
@kgodey I've marked this as a draft for now. Since #766 has some problems, I'd like to wait and see what happens to it before basing this PR off that work.

**Comment by seancolsen** [comment]
@pavish this is ready for review now

File: mathesar_ui/src/sections/import-data/preview/PreviewRows.svelte
Comment (seancolsen) [inline mathesar_ui/src/sections/import-data/preview/PreviewRows.svelte:12]: @pavish Is this import okay? Or do we want to add an alias and re-export all components from an `index.ts` file?

Comment (pavish) [reply mathesar_ui/src/sections/import-data/preview/PreviewRows.svelte:12]: We have a separate alias for the component library mainly because we want to keep changes to the minimal when we move it out into a separate package.

For mathesar specific components, I don't see a compelling need for an alias. So, I think this import is okay.

**Review by pavish — CHANGES_REQUESTED** [review]

File: mathesar_ui/src/constants.ts
Comment (seancolsen) [reply mathesar_ui/src/constants.ts]: Fixed in 8f529380b251861ec35f91e16e4a14d7460a04b6 as we discussed over chat

**Comment by seancolsen** [comment]
@pavish Ready for re-review

**Review by pavish — APPROVED** [review]
@seancolsen The changes look great to me!

**Pull request merged**
